### PR TITLE
[REF] Mark CRM_Contribute_BAO_Contribution_Utils::formatAmount deprec…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ated and replace its only usage with CRM_Utils_Rule::cleanMoney

Before
----------------------------------------
Non standard function used for formatting an amount for storing in the database

After
----------------------------------------
Standard function used

ping @eileenmcnaughton

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -5492,7 +5492,7 @@ public static function createProportionalEntry($entityParams, $eftParams) {
       $paid = $entityParams['line_item_amount'] * ($entityParams['trxn_total_amount'] / $entityParams['contribution_total_amount']);
     }
     // Record Entity Financial Trxn; CRM-20145
-    $eftParams['amount'] = CRM_Contribute_BAO_Contribution_Utils::formatAmount($paid);
+    $eftParams['amount'] = CRM_Utils_Rule::cleanMoney($paid);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php:5495]: No - we should remove the formatting. By the time it gets this deep it is formatted. cleanMoney actually corrupts the value if called twice on a Euro locale in some cases

Comment (seamuslee001) [reply CRM/Contribute/BAO/Contribution.php]: Wanted to let the test run so the tests passed with this change, now lets see if we rip it out what happens

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: Haven't you ripped out too much now?

Comment (seamuslee001) [reply CRM/Contribute/BAO/Contribution.php]: ah yes fixed but will be interested in that test run anyway

**Comment by seamuslee001** [comment]
@eileenmcnaughton does this work for you now?

**Comment by eileenmcnaughton** [comment]
Yes - looks good

**Pull request merged**